### PR TITLE
dev: listen on 127.0.0.1 in dev (INSECURE_DEV=true)

### PR DESCRIPTION
**Pull request by bobheadxi**
This PR sets `INSECURE_DEV` (maybe rename in a followup?) in `sg.config.yaml` by default, which has most services listening on `127.0.0.1` instead of all interfaces. Any services not yet configured on `INSECURE_DEV` has been updated to listen on `127.0.0.1` if the envvar is set.

Also removes all the MacOS firewall stuff in `sg`

It better close https://github.com/sourcegraph/sourcegraph/issues/29908 or I'm going to go nuts 😄 

## Test plan

```
git co docker-host
go run ./dev/sg start
```

### CODE REVIEW

File: sg.config.yaml [hunk 1]
```diff
@@ -6,6 +6,7 @@ env:
   PGDATABASE: sourcegraph
   PGSSLMODE: disable
   SG_DEV_MIGRATE_ON_APPLICATION_STARTUP: 'true'
+  INSECURE_DEV: true
```
Comment (varungandhi-src) [inline sg.config.yaml:9]: Calling people out like that, oof. 😮‍💨

File: cmd/frontend/envvar/envvar.go [hunk 2]
```diff
@@ -9,7 +9,12 @@ import (
...
 var HTTPAddrInternal = env.Get(
 	"SRC_HTTP_ADDR_INTERNAL",
-	"0.0.0.0:3090",
+	func() string {
+		if env.InsecureDev {
+			return "127.0.0.1:3090"
+		}
+		return "0.0.0.0:3090"
+	}(),
 	"HTTP listen address for internal HTTP API. This should never be exposed externally, as it lacks certain authz checks.",
```
Comment (bobheadxi) [inline cmd/frontend/envvar/envvar.go:18]: This is also used to redirect requests to frontend-internal by executor, Eric will test

Comment (efritz) [reply cmd/frontend/envvar/envvar.go:18]: Locally with this branch I got zero popups and was able to successfully auto-index a repository e2e.

**Review by efritz — APPROVED** [review]

**Review by abeatrix — APPROVED** [review]
I tried to look up a solution for this and got distracted, but this aligns with what I've found (use "[localhost" or "127.0.0.1" instead of "0.0.0.0"](https://apple.stackexchange.com/a/405967)) and running from this branch doesn't give me any pop-ups no more! Thank you so much for the fix  🚀

File: cmd/frontend/envvar/envvar.go (hunk 2 above)
Comment (bobheadxi) [reply cmd/frontend/envvar/envvar.go:18]: Thanks for checking!

**Pull request merged**